db-commons/src/main/scala/org/bitcoins/db/DbManagement.scala

```diff
@@ -1,34 +1,48 @@
 package org.bitcoins.db
 
+import org.bitcoins.core.util.BitcoinSLogger
 import org.flywaydb.core.Flyway
 import org.flywaydb.core.api.FlywayException
-import slick.jdbc.SQLiteProfile.api._
 
 import scala.concurrent.{ExecutionContext, Future}
 
-abstract class DbManagement extends DatabaseLogger {
-  def allTables: List[TableQuery[_ <: Table[_]]]
+trait DbManagement extends BitcoinSLogger {
+  _: JdbcProfileComponent[AppConfig] =>
+  import profile.api._
+
+  import scala.language.implicitConversions
+
+  /** Internally, slick defines the schema member as
+    *
+    * def schema: SchemaDescription = buildTableSchemaDescription(q.shaped.value.asInstanceOf[Table[_]])
+    *
+    * we need to cast between TableQuery's of specific table types to the more generic TableQuery[Table[_]]
+    * to get methods in this trait working as they require schema (which essentially does this cast anyway)
+    *
+    * This cast is needed because TableQuery is not covariant in its type parameter. However, since Query
+    * is covariant in its first type parameter, I believe the cast from TableQuery[T1] to TableQuery[T2] will
+    * always be safe so long as T1 is a subtype of T2 AND T1#TableElementType is equal to T2#TableElementType.
+    *
+    * The above conditions are always the case when this is called in the current code base and will
+    * stay that way so long as no one tries anything too fancy.
+    */
+  implicit protected def tableQueryToWithSchema(
+      tableQuery: TableQuery[_]): TableQuery[Table[_]] = {
+    tableQuery.asInstanceOf[TableQuery[Table[_]]]
+  }
+
+  def allTables: List[TableQuery[Table[_]]]
 
   /** Lists all tables in the given database */
-  def listTables(db: Database): Future[Vector[SQLiteTableInfo]] = {
+  def listTables: Future[Vector[SQLiteTableInfo]] = {
     import DbCommonsColumnMappers._
     val query = sql"SELECT * FROM sqlite_master where type='table'"
```

Comment:
Is there any way to extract `java.sql.DatabaseMetaData` using `slick`? This query will fail on many other databases.